bundle/edu.gemini.pot/src/main/scala/edu/gemini/spModel/inst/FeatureGeometry.scala

```diff
@@ -0,0 +1,161 @@
+   * @param guideStar the position of the guide star in arcseconds
+   * @return          the transformed shapes
+   */
+  def transformProbeArmForContext(shapes: List[Shape], armAngle: Double, guideStar: Point2D): List[Shape] = {
+    // For the guide star, we want to use the point closest to zero in terms of arcsec as a normalization.
+    val armTrans = AffineTransform.getRotateInstance(armAngle, guideStar.getX, guideStar.getY)
+    armTrans.concatenate(AffineTransform.getTranslateInstance(guideStar.getX, guideStar.getY))
+    shapes.map { armTrans.createTransformedShape }
+  }
+
+  /**
+   * Given a shape representing a guide probe arm or segment, an angle for the probe arm, and the position of the guide
+   * star, execute transformations on the shape to get the adjusted probe arm.
+   * @param shape     the shape to transform
+   * @param armAngle  the angle at which the probe arm will be situated
+   * @param guideStar the position of the guide star in arcsec
+   * @return          the transformed shapes
+   */
+  def transformProbeArmForContext(shape: Shape, armAngle: Double, guideStar: Point2D): Shape =
+    transformProbeArmForContext(List(shape), armAngle, guideStar).head
+
+  /**
+   * Given a list of geometry shapes, transform them as needed for display on the screen.
+   * @param shapes          the list of shapes to transform
+   * @param pixelsPerArcsec the pixel density per arcsec on the current display
+   * @return                the transformed shape
+   */
+  def transformProbeArmForScreen(shapes: List[Shape], pixelsPerArcsec: Double, xFlipArm: Boolean, flipRA: Double): List[Shape] = {
+    val xFlipFactor = if (xFlipArm) -1.0 else 1.0
+    val scaleTrans = AffineTransform.getScaleInstance(pixelsPerArcsec, pixelsPerArcsec)
+    scaleTrans.scale(flipRA, xFlipFactor)
+    shapes.map { scaleTrans.createTransformedShape }
+  }
+
+  /**
+   * Given a shape, transform it as needed for display on the screen.
+   * @param shape           the shape to transform
+   * @param pixelsPerArcsec the pixel density per arcsec on the current display
+   * @return                the transformed shape
+   */
+  def transformProbeArmForScreen(shape: Shape, pixelsPerArcsec: Double, xFlipArm: Boolean, flipRA: Double): Shape =
+    transformProbeArmForScreen(List(shape), pixelsPerArcsec, xFlipArm, flipRA).head
+
+  /**
+   * Given the parameters for a TPE representation of an offset, find the corresponding offset amongst an ObsContext's
+   * science positions.
+   *
+   *
+   */
+  def findObsContextOffset(ctx: ObsContext, ox: Double, oy: Double, bx: Double, by: Double, pixelsPerArcsec: Double): Option[Offset] = {
+    import scala.math.abs
+
+    // Convert from screen coordinates to (p,q) coordinates in arcseconds
+    val pq = List(bx - ox, by - oy).map(_ / pixelsPerArcsec)
+
+    // If the offset is (0,0), which is not included in the ObsContext's science positions, create a new Offset.
```

Comment:
Definitely you cannot rely on having an offset iterator in the sequence.  The intention is that there should always be at least one "science position" according to my reading of the code [here](https://github.com/gemini-hlsw/ocs/blob/develop/bundle/edu.gemini.pot/src/main/java/edu/gemini/spModel/target/offset/OffsetUtil.java#L123-L133) and [here](https://github.com/gemini-hlsw/ocs/blob/develop/bundle/edu.gemini.pot/src/main/java/edu/gemini/spModel/obs/context/ObsContext.java#L122-L129). This code is not great though and it wouldn't surprise me if you are right.  We should look into this if you can confirm that you're getting an `ObsContext` without science positions.  It'll probably bite us elsewhere otherwise.